internal/provider/bootstrap.go

```diff
@@ -48,9 +48,11 @@ func Bootstrap(e *pluggable.Event) pluggable.EventResponse {
 	tokenNotDefined := (p2pBlockDefined && prvConfig.P2P.NetworkToken == "") || !p2pBlockDefined
 	skipAuto := p2pBlockDefined && !prvConfig.P2P.Auto.IsEnabled()
 
-	node, _ := p2p.NewK8sNode(prvConfig)
-	if prvConfig.P2P == nil && node == nil {
-		return pluggable.EventResponse{State: fmt.Sprintf("no kubernetes distribution configuration. nothing to do: %s", cfg.Config)}
+	node, err := p2p.NewK8sNode(prvConfig)
+	if prvConfig.P2P != nil && err != nil {
+		if !strings.HasPrefix(err.Error(), "p2p is configured but no k8s component is explicitly enabled") {
```

Comment:
1. these 2 `if`s can be merged
2. we should rather check errors types or at least make the string a constant to avoid breaking it on the `NewK8sNode` side, if we just change the wording.